### DISCUSSION TITLE
Dynamic redirects based on user data

### DISCUSSION THREAD

**Discussion by Frost7994** [nextauthjs/next-auth#12124] [Help]
I’ve been trying this for a while now and I can’t seem to find a clean working solution. I want to redirect after signing a user in but the dashboard route depends on the users role and username which is stored in the session which I’ve configured in callbacks. So if a user signs in they’re redirected to /[role]/u/[username], both properties that live on the session.user. Does anyone have a solution?

**Comment by dev2xl** [comment]
Hello, Did you find a clean solution?
